### DISCUSSION TITLE
How to enable/disable opentelemetry tracing dynamically

### DISCUSSION THREAD

**Discussion by pankaj72981** [open-telemetry/opentelemetry-java-instrumentation#5972] [Q&A]
Dear folks,

How to enable/disable tracing dynamically without restarting the application service?

I tried to disable it dynamically using below APIs, it didnt work
GlobalOpenTelemetry.set(OpenTelemetry.noop());
GlobalOpenTelemetry.set(NoopOpenTelemetry.getInstance());

Any suggestion?

**Comment by jkwatson** [comment]
This isn't something that is currently supported. Are you using the javaagent, or just manually configuring your own SDK? If the latter, you could keep two copies around, and swap usages between the no-op instance and the fully configured instance. If you're using the javaagent, I'll move this issue over to the instrumentation repository, where that code lives.

**Comment by pankaj72981** [comment]
Thanks @jkwatson for the response, currently in Apache HBase we are using javaagent. It will be nice feature if Opentelemetry support dynamic enable/disable, so that it wont require process restart.

**Reply by jkwatson** [reply]
ok. I'll move this discussion over to the instrumentation repository

**Reply by ranveer091** [reply]
Do we have any suggestions now to achieve this ? @jkwatson @pankaj72981

**Reply by jkwatson** [reply]
I think this would require a feature request, but I'm not a maintainer on this project, so I don't know, really.

**Reply by mateuszrzeszutek** [reply]
Hey @ranveer091 ,
You should be able to dynamically change the sampling strategy during runtime by using the [remote Jaeger sampler](https://github.com/open-telemetry/opentelemetry-java/tree/main/sdk-extensions/jaeger-remote-sampler).
